bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqData.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.myq.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.commons.io.IOUtils;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonProcessingException;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import static org.openhab.io.net.http.HttpUtil.executeUrl;
+
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * This Class handles the Chamberlain myQ http connection.
+ * 
+ * <ul>
+ * <li>userName: myQ Login Username</li>
+ * <li>password: myQ Login Password</li>
+ * <li>sercurityToken: sercurityToken for API requests</li>
+ * <li>header: http header data</li>
+ * <li>webSite: url of myQ API</li>
+ * <li>appId: appId for API requests</li>
+ * </ul>
+ * 
+ * @author Scott Hanson
+ * @author Dan Cunningham
+ * @since 1.8.0
+ */
+public class MyqData {
+	static final Logger logger = LoggerFactory.getLogger(MyqData.class);
+
+	private static final String WEBSITE = "https://myqexternal.myqdevice.com";
+	public static final String DEFAULT_APP_ID = "JVM/G9Nwih5BwKgNCjLxiFUQxQijAebyyg8QUHr7JOrP+tuPb8iHfRHKwTmDzHOu";
+	public static final int DEFAUALT_TIMEOUT = 5000;
+
+	private String userName;
+	private String password;
+	private String appId;
+	private int timeout;
+	
+	private String sercurityToken;
+	private Properties header;
+
+	/**
+	 * Constructor For Chamberlain MyQ http connection
+	 * 
+	 * @param username
+	 *            Chamberlain MyQ UserName
+	 * 
+	 * @param password
+	 *            Chamberlain MyQ password
+	 * 
+	 * @param appId
+	 *            Chamberlain Application Id, defaults to DEFAULT_APP_ID if null
+	 * 
+	 * @param timeout
+	 *            HTTP timeout in milliseconds, defaults to DEFAUALT_TIMEOUT if not > 0
+	 */
+	public MyqData(String username, String password, String appId, int timeout) {
+		this.userName = username;
+		this.password = password;
+
+		if (appId != null) {
+			this.appId = appId;
+		} else {
+			this.appId = DEFAULT_APP_ID;
+		}
+
+		if (timeout > 0) {
+			this.timeout = timeout;
+		} else {
+			this.timeout = DEFAUALT_TIMEOUT;
+		}
+
+		header = new Properties();
+		header.put("Accept", "application/json");
+		header.put("User-Agent", "myq-openhab-api/1.0");
+	}
+
+	/**
+	 * Retrieves garage door device data from myq website, throws if connection
+	 * fails or user login fails
+	 * 
+	 */
+	public GarageDoorData getGarageData() throws InvalidLoginException,
+			IOException {
+		logger.debug("Retreiveing door data");
+		String url = String.format(
+				"%s/api/v4/userdevicedetails/get?appId=%s&SecurityToken=%s",
+				WEBSITE, appId, getSecurityToken());
```

Comment:
Shouldn't `appId` and `SecurityToken` values be URL-encoded (or any URL parameters that could potentially contain illegal characters elsewhere in this source file)?  I see the default app ID has `/` and `+` in them, for example.  The appId might be only accidentally working at present.